flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/tier/disk/DiskCacheManager.java

```diff
@@ -0,0 +1,157 @@
+        subpartitionCacheManagers[subpartitionId].appendEndOfSegmentEvent(record, dataType);
+        flushAndReleaseCacheBuffers();
+    }
+
+    /**
+     * Append buffer to {@link DiskCacheManager}.
+     *
+     * @param buffer to be managed by this class.
+     * @param subpartitionId the subpartition of this record.
+     */
+    public void append(Buffer buffer, int subpartitionId) {
+        subpartitionCacheManagers[subpartitionId].append(buffer);
+    }
+
+    /**
+     * Return the finished buffer index.
+     *
+     * @param subpartitionId the target subpartition id
+     * @return the finished buffer index
+     */
+    public int getFinishedBufferIndex(int subpartitionId) {
+        return subpartitionCacheManagers[subpartitionId].getFinishedBufferIndex();
+    }
+
+    /** Close this {@link DiskCacheManager}, it means no data can append to memory. */
+    public void close() {
+        flushAndReleaseCacheBuffers();
+    }
+
+    /**
+     * Release this {@link DiskCacheManager}, it means all memory taken by this class will recycle.
+     */
+    public void release() {
+        Arrays.stream(subpartitionCacheManagers).forEach(SubpartitionDiskCacheManager::release);
+        partitionFileWriter.release();
+    }
+
+    // ------------------------------------
+    //           Internal Method
+    // ------------------------------------
+
+    private void notifyFlushCachedBuffers() {
+        flushBuffers(false);
+    }
+
+    private void flushAndReleaseCacheBuffers() {
+        flushBuffers(true);
+    }
+
+    /**
+     * Note that the request of flushing buffers may come from the disk check thread or the task
+     * thread, so the method itself should ensure the thread safety.
+     */
+    private synchronized void flushBuffers(boolean needForceFlush) {
+        if (!needForceFlush && !hasFlushCompleted.isDone()) {
+            return;
+        }
+        List<SubpartitionNettyPayload> toWriteBuffers = new ArrayList<>();
+        int numToWriteBuffers = getSubpartitionBuffersToFlush(toWriteBuffers);
+
+        if (numToWriteBuffers > 0) {
+            CompletableFuture<Void> flushCompletableFuture =
+                    partitionFileWriter.write(toWriteBuffers);
+            if (!needForceFlush) {
+                hasFlushCompleted = flushCompletableFuture;
+            }
+        }
+    }
+
+    private int getSubpartitionBuffersToFlush(List<SubpartitionNettyPayload> toWriteBuffers) {
+        int numToWriteBuffers = 0;
+        for (int subpartitionId = 0; subpartitionId < numSubpartitions; subpartitionId++) {
+            List<NettyPayload> nettyPayloads =
+                    subpartitionCacheManagers[subpartitionId].getAllBuffers();
+            toWriteBuffers.add(
+                    new SubpartitionNettyPayload(
+                            subpartitionId,
+                            Collections.singletonList(
+                                    new SegmentNettyPayload(-1, nettyPayloads, false))));
```

Comment:
Why is the hardcoded `-1` segment id and `false` for finishing segment?

Reply:
Replaced the `-1` by the right segment id.